types/defines/pipelines.d.ts

```diff
@@ -2,12 +2,11 @@
 // Licensed under the Apache 2.0 license found in the LICENSE file or at:
 //     https://opensource.org/licenses/Apache-2.0
 
-export interface Pipeline {
+export interface Pipeline<T extends any> {
   /**
-   * send takes an array of javascript objects which are
-   * then received by the pipeline for processing
+   * The Pipeline interface represents the type of a binding to a Pipeline
    *
-   * @param data The data to be sent
+   * @param records The records to send to the pipeline
    */
-  send(data: object[]): Promise<void>
+  send(records: T[]): Promise<void>
```

Comment:
We can drop the generic and probably do `PipelineRecord` here?

```suggestion
  send(records: PipelineRecord[]): Promise<void>
```

Reply:
Should probably be `Pipeline<T extends PipelineRecord>`.  Seems like we want a way to allow the end implementor to enforce type safety in their code.  So if they specify for their environment:

```
interface MyEnv {
   MYPIPE: Pipeline<MyRecord>
}
```

It will enforce the shape of the send method at the call site:

```
  env.MYPIPE.send([{...}])
```